### ISSUE TITLE
DBContext.DeleteAsync throwing exception starting from v21.0.2

### ISSUE THREAD

**Issue by Germandf** [dj-nitehawk/MongoDB.Entities#198]
Here is how i save the entity:

``` csharp
await _dbContext.SaveAsync(entityInstance);
```

And how i delete it:

``` csharp
await _dbContext.DeleteAsync<Entity>(id);
```

Until `v21.0.1` it was ok, never changed the lines. Once I updated to `v21.0.2` it started to throw the following exception:

```
System.InvalidOperationException: ReadName can only be called when State is Name, not when State is EndOfDocument.
   at MongoDB.Bson.IO.BsonReader.ThrowInvalidState(String methodName, BsonReaderState[] validStates)
   at MongoDB.Bson.IO.BsonBinaryReader.ReadName(INameDecoder nameDecoder)
   at MongoDB.Bson.IO.IBsonReaderExtensions.VerifyName(IBsonReader reader, String expectedName)
   at MongoDB.Bson.IO.IBsonReaderExtensions.ReadName(IBsonReader reader, String name)
   at MongoDB.Driver.Linq.Linq3Implementation.Serializers.WrappedValueSerializer`1.Deserialize(BsonDeserializationContext context, BsonDeserializationArgs args)
   at MongoDB.Bson.Serialization.IBsonSerializerExtensions.Deserialize[TValue](IBsonSerializer`1 serializer, BsonDeserializationContext context)
   at MongoDB.Driver.Core.Operations.CursorBatchDeserializationHelper.DeserializeBatch[TDocument](RawBsonArray batch, IBsonSerializer`1 documentSerializer, MessageEncoderSettings messageEncoderSettings)
   at MongoDB.Driver.Core.Operations.FindOperation`1.CreateFirstCursorBatch(BsonDocument cursorDocument)
   at MongoDB.Driver.Core.Operations.FindOperation`1.CreateCursor(IChannelSourceHandle channelSource, IChannelHandle channel, BsonDocument commandResult)
   at MongoDB.Driver.Core.Operations.FindOperation`1.ExecuteAsync(RetryableReadContext context, CancellationToken cancellationToken)
   at MongoDB.Driver.Core.Operations.FindOperation`1.ExecuteAsync(IReadBinding binding, CancellationToken cancellationToken)
   at MongoDB.Driver.OperationExecutor.ExecuteReadOperationAsync[TResult](IReadBinding binding, IReadOperation`1 operation, CancellationToken cancellationToken)
   at MongoDB.Driver.MongoCollectionImpl`1.ExecuteReadOperationAsync[TResult](IClientSessionHandle session, IReadOperation`1 operation, ReadPreference readPreference, CancellationToken cancellationToken)
   at MongoDB.Driver.MongoCollectionImpl`1.UsingImplicitSessionAsync[TResult](Func`2 funcAsync, CancellationToken cancellationToken)
   at MongoDB.Entities.DB.DeleteAsync[T](FilterDefinition`1 filter, IClientSessionHandle session, CancellationToken cancellation, Collation collation)
```

I tried using `v21.0.2.4-beta` as suggested in #197 but the problem persists. Had to rollback to `v21.0.1`.

**Comment by dj-nitehawk** [comment]
apparently it looks like a bug in the mongo driver: https://jira.mongodb.org/browse/CSHARP-4538
so we will have to wait until they release the fixed driver.
will push a new beta out as soon as they do...
thanks!

**Comment by dj-nitehawk** [comment]
this should now be fixed in `v21.0.2.5-beta` with the latest version of the mongo driver.

do note there may be breaking changes in this beta. see changelog here: https://github.com/dj-nitehawk/MongoDB.Entities/blob/master/changelog.md
